test/src/test/java/jenkins/security/BasicHeaderProcessorTest.java

```diff
@@ -40,42 +54,57 @@ public void testVariousWaysToCall() throws Exception {
         User bar = User.get("bar");
 
         wc = j.createWebClient();
-
+    
         // call without authentication
         makeRequestWithAuthAndVerify(null, "anonymous");
 
+        // anonymous does not trigger any event
+
         // call with API token
         ApiTokenProperty t = foo.getProperty(ApiTokenProperty.class);
         final String token = t.getApiToken();
         makeRequestWithAuthAndVerify("foo:"+token, "foo");
+        //TODO verify why there are two events "authenticated" that are triggered
+        // the whole authentication process seems to be done twice
```

Comment:
Occurring only when testing.
In case of regular web login (password), rest call (pass / token), only one call is done.